### DISCUSSION TITLE
How to add custom fields to User model?

### DISCUSSION THREAD

**Discussion by andriusign** [directus/directus#3609] [Archive]
Hello. First of all I would like to thank for such a great solution as Directus. It is truly a great headless CMS that I was looking for so long. I have a question regarding some custom fields in User's model (_directus_users_ table). Tried searching the discussions here but could not find an answer. I want to add additional field to a User model which resides under directus_users table. I do not see how this is possible via back-end administrator's page, system tables (model) are not displayed. I want my user to have "username" field which would be filled in upon registration or email verification. The reason behind this is that I cannot display my registered user's email to public due to privacy, and setting up additional relational model only for this small addition does not make sense. Is there a way to add "username" field to User's model, and if so, how?

**Comment by rijkvanzanten** [comment]
@andriusign You can go to Settings > Data Model and in the right sidebar enable System Collections 🙂

**Comment by andriusign** [comment]
Ohh, found it! Sorry for being a noob :rofl: Thanks for the quick reply. After navigating to /admin/settings/data-model the right sidebar is auto-expanded, however, it did not occur to me to expand the Collection "accordion". +1 for you for hiding it. Maybe having an icon to expand accordion would improve UX like so:
![1](https://user-images.githubusercontent.com/14263959/103948788-04caf280-5143-11eb-8b16-dcaa23f203b1.png)
_vs_
![3](https://user-images.githubusercontent.com/14263959/103948697-e49b3380-5142-11eb-9acb-96edcf19c6dc.png)

BTW, I like that after reloading the page system collections are hidden again.

**Reply by rijkvanzanten** [reply]
>  it did not occur to me to expand the [...]

Fair enough! FWIW it's the same across the system, but I agree that a "expand" indicator on the right would be helpful here (cc @benhaynes)

> BTW, I like that after reloading the page system collections are hidden again.

The system collections are hidden by default, and it's open state isn't persisted, so this is expected behavior :)

**Reply by benhaynes** [reply]
Yeah, I like the expand icon idea... perhaps we can open a dedicated GitHub issue for tracking that?

**Reply by rijkvanzanten** [reply]
Ref https://github.com/directus/directus/issues/3621

**Comment by aypm-dev** [comment]
Just to add to the question, looks like newer versions of Directus show by default system collections in the data model page. And the option in the right sidebar is no longer there. Just saying this for anyone that has a good amount of collections (like me) and didn't notice the option (like me again)!

![image](https://user-images.githubusercontent.com/85044270/219558043-27084c0d-aaba-4134-a373-581629c809ab.png)
